cfe_internal/update/update_policy.cf

```diff
@@ -245,7 +245,7 @@ bundle agent cfe_internal_update_policy_cpv
 
     any::
 
-      "$(sys.workdir)$(const.dirsep)modules"
+      "$(inputs_dir)$(const.dirsep)modules"
```

Comment:
well, this is for the non-hubs right? thus `any::`

Reply:
I guess I'm wrong, the promise below would work for any validated policy. Would it already be validated on the agent if a new policy would be pulled down?

Reply:
Yes. I do think this is necessary. Otherwise updated modules won't get copied because they don't cause `validated_updates_ready` to be defined? I will look a bit into that class.

Reply:
@nickanderson  this `# - The class `validated_updates_ready` is defined when `cf_promises_validated` is repaired` in update policy.

Right, it is here in this files promise.

```cf3
      "$(inputs_dir)/cf_promises_validated"
      comment => "Check whether a validation stamp is available for a new policy update to reduce the distributed load",
      handle => "cfe_internal_update_policy_check_valid_update",
      copy_from => u_rcp("$(master_location)/cf_promises_validated", @(update_def.policy_servers)),
      action => u_immediate,
      classes => u_if_repaired("validated_updates_ready");
```

So if the copy in masterfiles/remote is different than the one in inputs then things will get copied.

We talked about this previously.

What do you think about leaving this chunk in or not?

Reply:
Yes, this chunk i believe should be left. It is the gating mechanism so that all clients are not hashing all files in inputs and perhaps more importantly so that the hub is not hashing all the files in masterfiles/remote for each of the clients every execution.